File: ruby/dbconncrud.rb
# Date: 2015-04-20
# Authore: Coding5
# Purpose: Ruby 建立DB CRUD 資料庫操作 
# Websit: http://www.coding5.com/works?transno=1000000013

require 'rubygems'
require 'mysql'

class DBConn
  def getConn
    @con = Mysql.new('localhost', 'username', 'passwd', 'DBNAME')
  end

  def runQuery(sql)
    @con.query(sql)
  end

  def runUpdate(sql)
    @con.query(sql)
  end

  def runInsert(sql)
    @con.query(sql)
  end

  def runInsert(sql)
    @con.query(sql)
  end

  def dbClose 
    @con.close
  end
end

File: python/dbconncrud.py
# Date: 2015-04-25
# Authore: Coding5
# Purpose: python 建立CRUD 操作function 
# Websit: http://www.coding5.com/works?transno=1000000019

import MySQLdb

class DBConn:
  def __init__(self):
    self.user = 'username'
    self.host = 'localhost'
    self.passwd = 'passwd'
    self.dbname = 'DBNAME'

  def dbConnect(self):
    self.db = MySQLdb.connect(
      self.host,self.user,self.passwd,self.dbname,charset='utf8')
    self.cursor = self.db.cursor()

  def runQuery(self, sql):
    self.cursor.execute(sql)
    self.results = self.cursor.fetchall()

  def runInsert(self, sql):
    self.cursor.execute(sql)
    self.db.commit()

  def runUpdate(self, sql):
    self.cursor.execute(sql)
    self.db.commit()

  def runDelete(self, sql):
    self.cursor.execute(sql)
    self.db.commit()

  def dbClose(self):
    self.db.close()
